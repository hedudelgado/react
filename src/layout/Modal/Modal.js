import React from 'react';
import {
  element, string, oneOfType, func,
} from 'prop-types';
import BaseStyle from '../BaseStyle';
import Button from '../Button';

import noop from '../../utils/noop';

const Modal = ({
  content,
  handleClose,
}) => (
  <BaseStyle component="div" className="pt3 wb-modal">
    <BaseStyle component="div" className="wb-modal-content">
      <Button
        data-test="close-modal"
        type="button"
        color="default"
        onClick={handleClose}
        className="wb-modal-content-close wb-btn wb-button-reset wb-icon_cross"
      />
      {content}
    </BaseStyle>
  </BaseStyle>
);

Modal.propTypes = {
  content: oneOfType([
    string,
    element,
  ]),
  handleClose: func,
};

Modal.defaultProps = {
  handleClose: noop,
};

export default Modal;
