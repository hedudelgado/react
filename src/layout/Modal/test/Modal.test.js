import React from 'react';
import { shallow } from 'enzyme';

import Modal from '../Modal';

jest.mock('../../../utils/noop', () => ({
  __esModule: true,
  default: jest.fn(),
}));

describe('Modal Component', () => {
  const content = 'content';
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Modal content={content} />);
  });

  describe('Rendering', () => {
    it('should show the component', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('handleClose', () => {
    it('should exist', () => {
      expect(Modal.defaultProps.handleClose).toBeDefined();
    });

    it('should be called when closing the modal', () => {
      const handleClose = jest.fn();
      const handleCloseWrapper = shallow(<Modal display handleClose={handleClose} />);

      handleCloseWrapper.find('[data-test="close-modal"]').simulate('click');

      expect(handleClose).toHaveBeenCalled();
    });
  });
});
