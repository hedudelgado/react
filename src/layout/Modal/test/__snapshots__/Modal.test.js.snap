// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Rendering should show the component 1`] = `
<BaseStyle
  className="pt3 wb-modal"
  component="div"
>
  <BaseStyle
    className="wb-modal-content"
    component="div"
  >
    <Button
      className="wb-modal-content-close wb-btn wb-button-reset wb-icon_cross"
      color="default"
      data-test="close-modal"
      fullWidth={false}
      href={null}
      onClick={[MockFunction]}
      submit={false}
      type="button"
      variant="button"
    />
    content
  </BaseStyle>
</BaseStyle>
`;
